lib/portlayer/metrics/subscription.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/docker/docker/pkg/pubsub"
+
+	"github.com/vmware/govmomi/object"
+
+	"github.com/vmware/vic/lib/portlayer/exec"
+)
+
+// subscription is a 1:1 relationship to a container and a
+// 1:M relationship to subscribers
+type subscription struct {
+	c *exec.Container
```

Comment:
Is it viable for this to use `vm.VirtualMachine` instead of exec.Container? I'm wondering if this needs to be in the portlayer or if it will happily sit in `pkg/vsphere`/potential inclusion into govmomi.